ibc-apps/ics20-transfer/types/src/denom.rs

```diff
@@ -293,19 +294,62 @@ pub fn is_receiver_chain_source(
 impl FromStr for PrefixedDenom {
     type Err = TokenTransferError;
 
+    /// Initializes a [`PrefixedDenom`] from a string that adheres to the format
+    /// `{port-id-1/channel-id-1}/{port-id-2/channel-id-2}/.../{port-id-n/channel-id-n}/base-denom`.
```

Comment:
One more thing. This fromat should get corrected:

```suggestion
    /// `{1st-port-id/channel-<index>}/{2nd-port-id/channel-<index>}/.../{Nth-port-id/channel-<index>}/<base denom>`.
```

Reply:
Even it should be reversed.

Reply:
063c1df6

Reply:
I think that it's fine with `1st-port-id`. As we have `TracePrefix::starts_with` naming.

Reply:
Maybe the sentence before should be adjusted: 
" _Initializes a [`PrefixedDenom`] from **a string that adheres to the format**_ "

Reply:
Hey, I'm not really sure if I understood you correctly. Can you please add a new suggestion for the text you want to replace?

Reply:
>vec![<nth-trace-prefix>, <n-1-th-trace-prefix>, ... <2nd-trace-prefix>, <1st-trace-prefix>]

Isn't opposite? We are pushing ([here](https://github.com/cosmos/ibc-rs/blob/063c1df6da3646915407d8e992c94f85c278a972/ibc-apps/ics20-transfer/types/src/denom.rs#L131)) / poping ([here](https://github.com/cosmos/ibc-rs/blob/063c1df6da3646915407d8e992c94f85c278a972/ibc-apps/ics20-transfer/types/src/denom.rs#L125)) the last element.

Then [this](https://github.com/cosmos/ibc-rs/blob/063c1df6da3646915407d8e992c94f85c278a972/ibc-apps/ics20-transfer/types/src/denom.rs#L94-L98) comment should be fixed as well.

Reply:
Similar comment in [ibc-go](https://github.com/cosmos/ibc-go/blob/0e3f428e66d6fc0fc6b10d2f3c658aaa5000daf7/modules/apps/transfer/types/trace.go#L188-L194).

Reply:
ah I understood it now. I was just counting it backwards. I can make it consistent with ibc-go :)

Reply:
@seanchen1991 let me take care of this.

Reply:
53fad12